rewrite-maven/src/test/java/org/openrewrite/maven/MavenParserTest.java

```diff
@@ -1267,9 +1262,7 @@ void activeByDefault() {
                         </profile>
                     </profiles>
                 </project>
-                """, spec -> {
-                  ((MavenParser.Builder) spec.getParser()).activeProfiles(activeProfile);
```

Comment:
When you remove this, it no longer seems to mimic what I see when I really try to use rewrite in this scenario.  In other words, the problematic code (or at least I think it's problematic:) is not exposed.  I totally get what you did in this change, and it's what I first tried.  However, when evaluation gets to where it determines if a profile is active, active profile is not `foobar`.  It's empty, which does not expose the issue.

<img width="601" alt="image" src="https://github.com/openrewrite/rewrite/assets/335662/66bbb397-b260-416c-8279-ee04effeccc0">

Reply:
Ah sorry about that! Thanks for keeping a sharp eye out. What's the current status of this PR now then? As I see a todo comment and draft status. Anything you'd like feedback on or plan to continue? Just so I know what to review/merge, or wait for.

Reply:
Hi @timtebeek .  I was kind of hoping you all would run with it, but I may be able to finish this up.  I'm just back from vacation.

Is it safe to assume that your team agrees this actually is a bug? :)